dll/win32/shell32/brfolder.cpp

```diff
@@ -757,6 +750,18 @@ BrFolder_OnInitDialog(HWND hWnd, BrFolder *info)
                                                  SHCNE_ALLEVENTS,
                                                  SHV_CHANGE_NOTIFY, 1, &ntreg);
 
+    if (!lpBrowseInfo->pidlRoot)
+    {
+        UINT csidl = (lpBrowseInfo->ulFlags & BIF_NEWDIALOGSTYLE) ? CSIDL_PERSONAL : CSIDL_DRIVES;;
+        if (LPITEMIDLIST pidl = SHCloneSpecialIDList(NULL, csidl, TRUE))
```

Comment:
No warning, this is perfectly correct code. If you declare the variable inside the if, then you don't use extra parentheses. If you are assigning `if ((x = y) != NULL)` then you must use extra parentheses and a comparison to avoid warnings.